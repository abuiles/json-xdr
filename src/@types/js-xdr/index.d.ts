// Type definitions for js-xdr v1.1.1
// Project: https://github.com/stellar/js-xdr
// Definitions by: Adolfo Builes <https://github.com/abuiles>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
// TypeScript Version: 3.4.1
declare module 'js-xdr' {
  import * as Long from 'long'

  export class Hyper extends Long {
  }

  export class UnsignedHyper extends Long {
  }

  export function config(fn: any, types?: any): any;
}
