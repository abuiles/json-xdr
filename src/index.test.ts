import { toJSON } from './index'
import * as XDR from 'js-xdr'

const types = XDR.config()

XDR.config((xdr) => {
  xdr.struct('Color', [
    ['version', xdr.int()],
    ['fee', xdr.uint()],
    ['fee', xdr.uint()],
    ['authorize', xdr.bool()],
    ['msg', xdr.string(11)],
    ['lat', xdr.float()],
    ['lon', xdr.double()],
    ['quadruple', xdr.quadruple()]
  ])
}, types)

describe('#toJSON', function() {
  test('converts XDR to JSON', () => {
    let color = new types.Color({
      version: -1,
      fee: 100,
      authorize: true,
      msg: 'hello world',
      lat: 37.7645352,
      lon: -122.421069,
      quadruple: 1.5
    })

    expect(toJSON(types, types.Color, color)).toMatchSnapshot()
  })
})
