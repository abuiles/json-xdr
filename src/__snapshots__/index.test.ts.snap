// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toJSON converts XDR to JSON 1`] = `
Object {
  "authorize": true,
  "fee": 100,
  "lat": 37.7645352,
  "lon": -122.421069,
  "msg": "hello world",
  "quadruple": 1.5,
  "version": -1,
}
`;
